R/transformCounts.R

```diff
@@ -308,6 +351,115 @@ setMethod("transformAssay", signature = c(x = "SummarizedExperiment"),
     return(mat)
 }
 
+####################################.calc_css###################################
+# This function applies cumulative sum scaling (CSS) to the abundance table.
+.calc_css <- function(mat, css.percentile = 0.5, scaling = 1000, ...) {
+    # Input check
+    if( !.is_integer(scaling) ){
+        stop("'scaling' must be an integer value.", call. = FALSE)
+    }
+    if( !is.numeric(css.percentile) ){
+        stop("'css.percentile' must be a numeric value.", call. = FALSE)
```

Comment:
This probably pass through vectors?

--> .is_an_integer and is.numeric() && length() == 1

Reply:
No, I meant that you should use .is_an_integer(scaling) and is.numeric(css.percentile) && length(css.percentile) == 1

Reply:
thanks.